application/shared-kernel/ApiCore/Middleware/WebAppMiddleware.cs

```diff
@@ -119,6 +127,23 @@ private string GetHtmlWithEnvironment(Dictionary<string, string>? requestEnviron
 
         return result;
     }
+
+    private static void ApplyNoCacheHeaders(HttpContext context)
+    {
+        context.Response.Headers.Append("Cache-Control", "no-cache, no-store, must-revalidate");
+        context.Response.Headers.Append("Pragma", "no-cache");
+    }
+
+    private static void ApplySecurityHeaders(HttpContext context, StringValues contentSecurityPolicy)
+    {
+        context.Response.Headers.Append("Content-Security-Policy", contentSecurityPolicy);
+        // Disable automatic content type detection
+        context.Response.Headers.Append("X-Content-Type-Options", "nosniff");
+        context.Response.Headers.Append("X-Frame-Options", "DENY");
+        context.Response.Headers.Append("X-XSS-Protection", "1; mode=block");
+        context.Response.Headers.Append("Strict-Transport-Security", "max-age=31536000; includeSubDomains");
```

Comment:
The HTTPS strict header is handled by Program.cs. I believe most if not alle of these should apply to both API and Index.html end-points.

Reply:
Ok - I can remove the ApplySecurityHeaders then

Reply:
I'm a bit in doubt. Maybe it's only "trict-Transport-Security". Not sure if it makes sense to see X-Frame-Options and X-XSS-Protection on API calls. Maybe these do belong here as they are only for HTML?

Reply:
I've removed it for now - I'm sure we will revisit this (I constantly look at lighthouse and security issues / audit in the browser)